packages/lib/services/survey.ts

```diff
@@ -73,188 +85,254 @@ export const selectSurveyWithAnalytics = {
-
-        const transformedSurvey = {
-          ...surveyPrisma,
-          triggers: surveyPrisma.triggers.map((trigger) => trigger.eventClass),
-          analytics: {
-            numDisplays,
-            responseRate,
-            numResponses: _count.responses,
+  // since the unstable_cache function does not support deserialization of dates, we need to manually deserialize them
+  // https://github.com/vercel/next.js/issues/51613
+  return surveys.map((survey) => ({
+    ...survey,
+    createdAt: new Date(survey.createdAt),
+    updatedAt: new Date(survey.updatedAt),
+  }));
+};
+
+export const getSurveysWithAnalytics = async (environmentId: string): Promise<TSurveyWithAnalytics[]> => {
+  const surveysWithAnalytics = await unstable_cache(
+    async () => {
+      validateInputs([environmentId, ZId]);
+      let surveysPrisma;
+      try {
+        surveysPrisma = await prisma.survey.findMany({
+          where: {
+            environmentId,
           },
-        };
-        const survey = ZSurveyWithAnalytics.parse(transformedSurvey);
-        surveys.push(survey);
+          select: selectSurveyWithAnalytics,
+        });
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
       }
-      return surveys;
-    } catch (error) {
-      if (error instanceof z.ZodError) {
-        console.error(JSON.stringify(error.errors, null, 2)); // log the detailed error information
+
+      try {
+        const surveys: TSurveyWithAnalytics[] = [];
+        for (const { _count, displays, ...surveyPrisma } of surveysPrisma) {
+          const numDisplays = displays.length;
+          const numDisplaysResponded = displays.filter((item) => item.status === "responded").length;
+          const responseRate = numDisplays ? Math.round((numDisplaysResponded / numDisplays) * 100) / 100 : 0;
+
+          const transformedSurvey = {
+            ...surveyPrisma,
+            triggers: surveyPrisma.triggers.map((trigger) => trigger.eventClass),
+            analytics: {
+              numDisplays,
+              responseRate,
+              numResponses: _count.responses,
+            },
+          };
+          const survey = ZSurveyWithAnalytics.parse(transformedSurvey);
+          surveys.push(survey);
+        }
+        return surveys;
+      } catch (error) {
+        if (error instanceof z.ZodError) {
+          console.error(JSON.stringify(error.errors, null, 2)); // log the detailed error information
+        }
+        throw new ValidationError("Data validation of survey failed");
       }
-      throw new ValidationError("Data validation of survey failed");
+    },
+    [getSurveysWithAnalyticsCacheKey(environmentId)],
+    {
+      tags: [getSurveysCacheTag(environmentId)], // TODO: add tags for displays and responses
```

Comment:
@pandeymangg I optimised the naming of the cache tags and fixed some issues with revalidation. Please take a look at the changes I made in the last commit.
One last thing that is open is the revalidation of the `getSurveysWithAnalytics`. This returns the list of all surveys in the environment together with the number of displays and responses for this surveys.
So the cache needs to revalidate when a survey is created/updated/deleted and if a new response or display is added/deleted in the environment.
The check for the surveys we already have, tricky is the number of responses/displays, as we don't have the environmentId in the responses/displays to revalidate on e.g. `deleteResponse`. Do you have an idea how we can solve this?
As an alternative, I think for now we could also change the frontend, so we don't list `surveysWithAnalytics`, but `surveys` (leave out the number of responses) ;-)

Reply:
Update: We will leave out the number of responses for now and render a list of TSurveys :-)